File: backend/src/main/java/unpsjb/labprog/backend/model/Usuario.java
package unpsjb.labprog.backend.model;

import org.springframework.data.neo4j.core.schema.*;

import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Data
@Getter
@Setter
@Node("Usuario")
@NoArgsConstructor
public class Usuario {

    @Id
    @GeneratedValue
    private Long id;

    private String nombreUsuario;   
    private String nombreReal;
    private LocalDate fechaNacimiento;
    private LocalDate fechaDeCreacion;
    private String correoElectronico;
    private String contrasena;
    private String genero;
    private String descripcion;
    private double latitud;
    private double longitud;
    private String privacidadPerfil;
    private String privacidadEventos;
    private String privacidadComunidades;
    private String fotoPerfil;
    private boolean borrarNotificacionesEventosPasados;


    
}

File: frontend/cli/src/app/usuarios/usuario.ts
export interface Usuario {
    id: number;
    nombreUsuario: string;
    nombreReal: string;
    fechaNacimiento: Date;
    fechaDeCreacion: Date;
    correoElectronico: string;
    contrasena: string;
    descripcion: string;
    latitud: number;
    longitud: number;
    fotoPerfil: string;
    privacidadPerfil?: 'Privada' | 'Solo amigos' | 'Pública';
    privacidadEventos?: 'Privada' | 'Solo amigos' | 'Pública';
    privacidadComunidades?: 'Privada' | 'Solo amigos' | 'Pública';
}
